### PR TITLE
fix get_metrics method

### PR DESCRIPTION
**Pull request by maypink**
Earlier in this method, when calculating ROCAUC metric, predictions were redefined and probabilities were placed in `prediction.predict`. Then the probabilities were converted into labels
Now at the very beginning when calculating the metric for which labels are needed, predictions are given in labels

### CODE REVIEW

File: test/unit/api/test_api_utils.py [hunk 1]
```diff
@@ -51,22 +51,17 @@ def test_compose_fedot_model_with_tuning():
...
-    assert int(prediction.predict[1]) == 1
+    task_type = 'classification'
+
+    train_data, test_data, _ = get_dataset(task_type=task_type)
+
+    model = Fedot(problem=task_type, timeout=0.5)
+    model.fit(train_data)
+    model.predict(test_data)
+    metrics = model.get_metrics(metric_names=['roc_auc', 'f1'])
+    assert isinstance(metrics['f1'], float)
```
Comment (Dreamlone) [inline test/unit/api/test_api_utils.py:64]: Лучше стоит как-нибудь более низкоуровнего это делать, а то сейчас для проверки небольшой части функциональности API (расчета метрик), запускается весь AutoML арсенал. Можно как-нибудь протестировать (кстати get_dataset(task_type=task_type) точно подходит для этого теста?) 

Сам тест кстати очень долгий вышел
image

Можно если что вот так задать 
`model.fit(train_data, predefined_model='logit')`
Работает пошустрее, так как обучает всего один пайплайн
image

**Review by Dreamlone — CHANGES_REQUESTED** [review]

File: test/unit/api/test_api_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (maypink) [reply test/unit/api/test_api_utils.py]: [done](https://github.com/nccr-itmo/FEDOT/pull/620/commits/7f678ad97dd3a3c54c02dca08d99d9e07c126c4a#diff-33b3da5de920cc340bbc6820f766f3d73a87d0545f26ca587bd630cd5cc8a8d3R81-R82)

Comment (maypink) [reply test/unit/api/test_api_utils.py]: [ну да, в ирисах действительно многовато для бинарности))](https://github.com/nccr-itmo/FEDOT/pull/620/commits/7f678ad97dd3a3c54c02dca08d99d9e07c126c4a#diff-33b3da5de920cc340bbc6820f766f3d73a87d0545f26ca587bd630cd5cc8a8d3R54)

Comment (Dreamlone) [reply test/unit/api/test_api_utils.py]: Можно ещё немного упростить

```Python
task = Task(TaskTypesEnum.classification)
data_processor = ApiDataProcessor(task=task)
```

И тогда не нужно целого Fedot инициализировать.

**Review by Dreamlone — APPROVED** [review]

**Review by nicl-nno — APPROVED** [review]

File: test/unit/api/test_api_utils.py (hunk 1 above)
Comment (maypink) [reply test/unit/api/test_api_utils.py]: кажется, что до get_metrics я таким способом не доберусь. мне же из оттуда нужен не только метод correct_predictions у data_processor, логику внутри тоже хотелось бы проверить

Comment (Dreamlone) [reply test/unit/api/test_api_utils.py]: Вообще да, согласен. Ну тогда стоит откатиться всё таки к первому варианту и просто добавить строку 
model.fit(train_data, predefined_model='logit')

Конструкции `model.data_processor.define_data` выглядят сильно громоздкими
